ports/atmel-samd/common-hal/pulseio/PulseIn.c

```diff
@@ -126,10 +150,62 @@ void common_hal_pulseio_pulsein_construct(pulseio_pulsein_obj_t* self,
     self->start = 0;
     self->len = 0;
     self->first_edge = true;
-    self->last_us = 0;
-    self->last_ms = 0;
     self->errored_too_fast = false;
 
+    if (refcount == 0) {
+        // Find a spare timer.
+        Tc *tc = NULL;
+        int8_t index = TC_INST_NUM - 1;
+        for (; index >= 0; index--) {
+            if (tc_insts[index]->COUNT16.CTRLA.bit.ENABLE == 0) {
+                tc = tc_insts[index];
+                break;
+            }
+        }
+        if (tc == NULL) {
+            mp_raise_RuntimeError(translate("All timers in use"));
+        }
+
+        pulsein_tc_index = index;
+
+        set_timer_handler(true, index, TC_HANDLER_PULSEIN);
+        #ifdef SAMD21
+        // We use GCLK0 for SAMD21 which is 48mhz. We prescale it to 3mhz.
```

Comment:
This is very pedantic: "Hz" is capitalized. "mHz" is millhertz (like milliamps), "MHz" is megahertz.